nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -267,6 +276,22 @@ in
       '';
     };
 
+    enableReload = mkOption {
+      default = true;
+      type = types.bool;
+      description = lib.mdDoc ''
+        Reload Caddy when configuration file changes (instead of restart).
+        The configuration file is exposed at {file}`/etc/${etcConfigFile}`.
+
+        Note that enabling this option requires the [admin API](https://caddyserver.com/docs/caddyfile/options#admin)
```

Comment:
Can we assert that?

Reply:
I guess, technically, we could run [`caddy adapt`](https://caddyserver.com/docs/command-line#caddy-adapt) in a `pkgs.runCommand` and check if [`admin.disabled`](https://caddyserver.com/docs/json/admin/) in the resulting json is either unset or set to `false`.

Other than that? No, I don't think so.

Reply:
I’ve added an assertion for that. Reading the whole adapted config with `readFile` and `fromJSON` resulted in an error saying I can’t use a file that contains other derivations paths, so I used `jq` to only extract the interesting part. Let me know if there’s a better way to do it.

Reply:
we could now drop this note, but we can also keep it.